giraf/binaryGiraf/binReader.go

```diff
@@ -0,0 +1,269 @@
+package binaryGiraf
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"github.com/biogo/hts/bgzf"
+	"github.com/vertgenlab/gonomics/cigar"
+	"github.com/vertgenlab/gonomics/common"
+	"github.com/vertgenlab/gonomics/dna"
+	"github.com/vertgenlab/gonomics/dnaThreeBit"
+	"github.com/vertgenlab/gonomics/fileio"
+	"github.com/vertgenlab/gonomics/giraf"
+	"github.com/vertgenlab/gonomics/simpleGraph"
+	"io"
+	"log"
+	"strings"
+)
+
+// The BinReader struct wraps the bgzf reader from the biogo repository with a bytes buffer to store encoded giraf records
+type BinReader struct {
+	bg       *bgzf.Reader
+	currData *bytes.Buffer
+}
+
+// NewBinReader creates a new BinReader
+func NewBinReader(file io.Reader) *BinReader {
+	reader, err := bgzf.NewReader(file, 1) //TODO: Play with different levels of concurrency
```

Comment:
My intuition tells me that this would not get that much faster when you increase this number since reading in a file will we bounded by IO not CPU. If i understand this correctly what I think ends up happening is you spend time setting up the goroutines and they sit around and don't have work to do because there are more CPU operations than IO processes. What you can do is have one thread help out with the decompression and the other thread read the file which yields to at most 2x speed up, but not much more.

My take on bgzf reader is that it implements really cool multi-threading functionality, but concurrency/ parallelism is only beneficial if your function takes longer than setting up your channels and goroutines and the calls to sync to make your processes thread safe. Base on my intuition, I think concurrency/ parallelism here would either slow down reading process (or remain the same in terms of time) and for sure increase the memory footprint because of the additional goroutines that are initialized only to sit around and do almost nothing.